lib/polyfill/media_capabilities.js

```diff
@@ -39,14 +39,16 @@ shaka.polyfill.MediaCapabilities = class {
       navigator.mediaCapabilities) {
       shaka.log.debug(
           'MediaCapabilities: Native mediaCapabilities support found.');
-      return;
-    }
-
-    if (!navigator.mediaCapabilities) {
-      navigator.mediaCapabilities = /** @type {!MediaCapabilities} */ ({});
+    } else {
+      if (!navigator.mediaCapabilities) {
+        navigator.mediaCapabilities = /** @type {!MediaCapabilities} */ ({});
+      }
+      navigator.mediaCapabilities.decodingInfo =
+          shaka.polyfill.MediaCapabilities.decodingInfo_;
     }
-    navigator.mediaCapabilities.decodingInfo =
-        shaka.polyfill.MediaCapabilities.decodingInfo_;
+    // Workaround because in Apple browsers there is no way to overwrite
+    // MediaCapabilities.
+    window.shakaMediaCapabilities = navigator.mediaCapabilities;
```

Comment:
If this works, doesn't that imply that setting the value above on line 44 was successful?  That would seem to contradict the need for this PR.

Reply:
If I recall, when I last looked into this issue it looked like what was happening was that `navigator.mediaCapabilities` was being set successfully, but then being reset to the default value in some later stage of loading... I wasn't able to quite figure out what was resetting it, though.
Storing it into a custom variable would let us work around it being reset, basically.

Reply:
Since all polyfills are now exported, it seems appropriate to keep this. Also this works if called at a later stage.

Reply:
I'm going to experiment a little to see if I can understand the root cause before merging.

Reply:
I'm not able to reproduce this yet with DASH content.  Does the MCap polyfill only get replaced if I play FairPlay content?

Reply:
With the content you sent us via email, I am still unable to reproduce the issue.  (Safari 14.1.1)  I see MediaCapabilities negotiating successfully and instantiating a FairPlay CDM.  (The license request fails with an HTTP 500 error, but at least MCap is successful.)

Reply:
Let's move conversation to https://github.com/google/shaka-player/issues/3530#issuecomment-933767346